packages/config/src/class-config.php

```diff
@@ -197,4 +250,62 @@ protected function enable_connection() {
 		return true;
 	}
 
+	/**
+	 * Setup the Connection options.
+	 */
+	protected function ensure_options_connection() {
+		$options = $this->get_feature_options( 'connection' );
+
+		if ( ! empty( $options['slug'] ) ) {
+			add_action(
+				'jetpack_connection_configure_plugin',
+				function() use ( $options ) {
+					if ( empty( $options['slug'] ) ) {
+						return;
+					}
+
+					$slug = $options['slug'];
+					unset( $options['slug'] );
+
+					$name = $slug;
+					if ( ! empty( $options['name'] ) ) {
+						$name = $options['name'];
+						unset( $options['name'] );
+					}
+
+					( new Plugin( $slug ) )->add( $name, $options );
```

Comment:
Isn't it a good idea to add a verification here if the plugin was successfully added?

Reply:
If anything goes wrong, an error should be triggered inside the `connection` package.
I don't think we can do anything at the config class. Even if the plugin can't be added, we should continue the execution.